File: src/main/java/io/gatling/custom/browser/javaapi/actions/BrowserClearContext.java
package io.gatling.custom.browser.javaapi.actions;

import io.gatling.custom.browser.actions.builder.BrowserClearActionsBuilder;
import io.gatling.javaapi.core.ActionBuilder;

public class BrowserClearContext implements ActionBuilder {

    private final BrowserClearActionsBuilder wrapped;

    public BrowserClearContext(BrowserClearActionsBuilder wrapped) {
        this.wrapped = wrapped;
    }

    @Override
    public io.gatling.core.action.builder.ActionBuilder asScala() {
        return wrapped;
    }
}

File: src/main/scala/io/gatling/custom/browser/actions/BrowserClearContext.scala
package io.gatling.custom.browser.actions

import io.gatling.core.action.{Action, ChainableAction}
import io.gatling.core.session.Session
import io.gatling.core.stats.StatsEngine
import io.gatling.core.structure.ScenarioContext
import io.gatling.core.util.NameGen
import io.gatling.custom.browser.actor.{BrowserCommandResponse, BrowserWorkerActor}
import io.gatling.custom.browser.utils.Constants

import scala.concurrent.Await

case class BrowserClearContext(ctx: ScenarioContext, next: Action) extends ChainableAction with NameGen with BrowserActionsBase {

  override def statsEngine: StatsEngine = ctx.coreComponents.statsEngine

  override def name: String = genName("clearBrowserContext")

  override protected def execute(session: Session): Unit = {

    val userId = session.userId
    logger.trace(s"userID-$userId, execute ClearContext action")

    if(browserActorPool.existActor(userId)){

      val browserActorWorker = browserActorPool.getActorById(userId)
      val promise = browserActorWorker.replyPromise[Unit](Constants.PROMISE_TIMEOUT)

      browserActorWorker ! BrowserWorkerActor.ClosePage(promise)
      Await.result(promise.future, Constants.PROMISE_TIMEOUT)
      logger.trace(s"userID-$userId, remove BrowserContext from BrowserWorkerActor")
    }
    next ! session
  }
}
